### PR TITLE
JAVAFICATION: Dry up and move pipelines to Java in part

### PR DESCRIPTION
**Pull request by original-brownbear**
Extracted shared logic between Ruby and Java pipeline to common base class implemented in Java.

* No behaviour changes to outside APIs
* Had to turn some internal instance variable lookups with `@variable` to getter calls because defining instance variables from Java is kinda awkward and hard to follow
* CommonsCodec  was added to get the `Hex.toHexString` functionality

### CODE REVIEW

**Comment by andrewvc** [comment]
jenkins, please retest this.

**Comment by andrewvc** [comment]
some interesting test failures, but I think they're unrelated. I'd like to wait for a passing run however:

```
21:29:22     org.jruby.exceptions.RaiseException: (GemNotFound) Could not find rake-12.3.1 in any of the sources
```

Also, I notice these, but I don't think they're related:

```

21:29:11 /opt/logstash/logstash-core/src/test/java/org/logstash/EventTest.java:102: error: unmappable character for encoding ASCII
21:29:11         e.setField("foo", "b??r");
21:29:11                             ^
21:29:11 /opt/logstash/logstash-core/src/test/java/org/logstash/EventTest.java:102: error: unmappable character for encoding ASCII
21:29:11         e.setField("foo", "b??r");
```

**Comment by original-brownbear** [comment]
@andrewvc hmm other PRs (like this doc one https://logstash-ci.elastic.co/job/elastic+logstash+pull-request/709/) fail the exact same way. Seems we have some issue with Jenkins in general ...

**Comment by original-brownbear** [comment]
Jenkins test this

**Comment by original-brownbear** [comment]
Build failures are fixed in https://github.com/elastic/logstash/pull/9685
